test/route.test.js

```diff
@@ -173,7 +173,7 @@ test('route', t => {
 test('invalid schema - route', t => {
   t.plan(3)
 
-  const fastify = Fastify()
+  const fastify = Fastify({ exposeHeadRoutes: false })
```

Comment:
I set `exposeHeadRoutes` to false in order to let the test pass with the correct route, as now if there's a schema validation error, the `HEAD` route validation is being thrown first, just before the origin `GET` route

Reply:
Have you rebased this PR on top of `next`? I think I fixed this.

Reply:
Yes, actually rebasing to `next` popped up the issue. The routes are registered in order, first `GET` and `HEAD` as second, but as the schema build happens async, the first that comes up is always for the `HEAD` one

Reply:
I have a feeling that we should fix this.

Reply:
I found the issue, basically in order to receive first the error thrown by the schema validation of the first `GET` route, is a need to pass the error to the `done` function when receiving the error from the schema validation. Solved on 1ac09cc